ext/lib/crypto/mbedtls/configs/config-coap.h

```diff
@@ -91,15 +91,11 @@
 #define MBEDTLS_SSL_CIPHERSUITES  MBEDTLS_TLS_PSK_WITH_AES_128_CCM_8
 
 /*
- * Save RAM at the expense of interoperability: do this only if you control
- * both ends of the connection!  (See comments in "mbedtls/ssl.h".)
+ * Allow to save RAM at the expense of interoperability: do this only if you
+ * control both ends of the connection!  (See comments in "mbedtls/ssl.h".)
  * The optimal size here depends on the typical size of records.
  */
-#ifndef CONFIG_COAP_MBEDTLS_SSL_MAX_CONTENT_LEN
-#define MBEDTLS_SSL_MAX_CONTENT_LEN	1500
-#else
-#define MBEDTLS_SSL_MAX_CONTENT_LEN	CONFIG_COAP_MBEDTLS_SSL_MAX_CONTENT_LEN
-#endif
+#define MBEDTLS_SSL_MAX_CONTENT_LEN  CONFIG_MBEDTLS_SSL_MAX_CONTENT_LEN
```

Comment:
I know we have CONFIG_MBEDTLS_BUILTIN=y all over our sample .conf files at the moment, but CONFIG_MBEDTLS_SSL_MAX_CONTENT_LEN depends on CONFIG_MBEDTLS_BUILTIN and it's still a manual setting.  Do we want to remove the #ifndef here, or merely change it to CONFIG_MBEDTLS_SSL_MAX_CONTENT_LEN?

Reply:
Well, my idea is that people either: a) use CONFIG_MBEDTLS_BUILTIN=y, choose one of (hopefully growing number of) Zephyr-provided mbedTLS configs, and are beyond that, able to adjust "interesting" options using Zephyr Kconfig options, or b) use CONFIG_MBEDTLS_BUILTIN=n and used both mbedTLS code and config coming from outside Zephyr. IMHO, that's reasonable.

If you have concerns specifically about config-coap.h, I can put ifdef back.

Reply:
Is there a case for people to use a Zephyr provided config (like coap.h) but building from an external source.  If we don't think we should be supporting that then I'm ok with it as is.  If not, we probably need the #ifndef back.

Reply:
My opinion is like above - it makes sense to support "polar" options, either fully integrated into Zephyr, or fully external. I guess people would use external option to retain full control of how mbedTLS is built. Using Z's config won't allow that, as Z's config may evolve with it ("unexpectedly" to users). Then trying to support various mix-in and half-way options only leads to extra effort (all those ifdef's) and confusion.

Anyway, that's just my opinions. @jukkar, @tbursztyka, @nashif : Any thoughts on the discussion above?

Reply:
@pfalcon thanks for the discussion.  I'm ok w/ leaving it as is.